whittle/metrics/latency.py

```diff
@@ -0,0 +1,93 @@
+import torch
+import torch.profiler
+from torch.profiler import record_function
+
+
+# Define a sample language model (e.g., a simple RNN or transformer-based model)
+def get_total_cpu_gpu_runtime(prof):
+    # from torch.autograd.profiler_utils.py (_build_table)
+    # results is in micro-seconds
+    from torch.autograd import DeviceType
+
+    events = prof.events()
+    sum_self_cpu_time_total = sum([event.self_cpu_time_total for event in events])
+    sum_self_cuda_time_total = 0
+    for evt in events:
+        if evt.device_type == DeviceType.CPU:
+            # in legacy profiler, kernel info is stored in cpu events
+            if evt.is_legacy:
+                sum_self_cuda_time_total += evt.self_cuda_time_total
+        elif evt.device_type == DeviceType.CUDA:
+            # in kineto profiler, there're events with the correct device type (e.g. CUDA)
+            sum_self_cuda_time_total += evt.self_cuda_time_total
+    return sum_self_cpu_time_total, sum_self_cuda_time_total
+
+
+def profile_model_latency(model, use_cuda=False, batch_size=8, n_samples=10):
```

Comment:
Maybe for  a future PR: We are a bit inconsistent with the naming for metrics. Right now we have `compute_parameters` , `profile_model_latency` and `weight_magnitude`. We should unify this such that they all follow the same pattern.

Reply:
Will do that in a separate PR

Reply:
added an issue #142